packages/expo/src/Linking/LinkingModule.web.ts

```diff
@@ -0,0 +1,67 @@
+import findIndex from 'array-find-index';
+import invariant from 'invariant';
+
+// TODO: Bacon: Is there enough coverage to use Array.prototype.findIndex() directly: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex#Browser_compatibility
+
+type NativeEvent = { data: any; origin: string; source: any };
+type EventType = { url: string; nativeEvent: NativeEvent };
+type HandlerFunction = (event: EventType) => {};
+
+// TODO: Bacon: Should we alias url to `hashchange` or `popstate`: https://stackoverflow.com/a/44318564/4047926
+
+const EventTypes = ['url'];
+
+const messageEventKey = 'message';
+
+const listeners: Array<[HandlerFunction, any]> = [];
+
+function _validateURL(url: string): void {
+  invariant(typeof url === 'string', `Invalid URL: should be a string. Instead found: ${url}`);
+  invariant(url, 'Invalid URL: cannot be empty');
+}
+
+// For better parity this should extend EventEmitter like React Native
+class Linking {
+  constructor() {}
+
+  addEventListener(type: 'url', handler: HandlerFunction): void {
+    invariant(
+      EventTypes.indexOf(type) !== -1,
+      `Linking.addEventListener(): ${type} is not a valid event`
+    );
+    const callback = nativeEvent => handler({ url: window.location.href, nativeEvent });
+    listeners.push([handler, callback]);
+    window.addEventListener(messageEventKey, callback, false);
```

Comment:
this should probably listen to popstate and perhaps hashchange, not message

Reply:
I think the object that is returned is closer to what is returned in RN. I also added a screen to NCL so we could test functionality.